server/stream.go

```diff
@@ -4716,10 +4716,19 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 			// Assume no interest and check to disqualify.
 			noInterest = true
 			mset.clsMu.RLock()
+		out:
```

Comment:
This could have an impact if large number of consumers or large number of filter subjects. We already have these leaded into a sublist, so let's use that.

```go
noInterest := !mset.csl.hasInterest(subject)
```

Reply:
That nicely simplifies it, done!